parsers/ocaml.c

```diff
@@ -25,39 +25,43 @@
  * -Wextra */
 #ifdef UNUSED
 #elif defined(__GNUC__)
-# define UNUSED(x) UNUSED_ ## x __attribute__((unused))
+#define UNUSED(x) UNUSED_##x __attribute__ ((unused))
 #elif defined(__LCLINT__)
-# define UNUSED(x) /*@unused@*/ x
+#define UNUSED(x) /*@unused@*/ x
 #else
-# define UNUSED(x) x
+#define UNUSED(x) x
 #endif
 #define OCAML_MAX_STACK_SIZE 256
 
-typedef enum {
-	K_CLASS,        /* Ocaml class, relatively rare */
-	K_METHOD,       /* class method */
-	K_MODULE,       /* Ocaml module OR functor */
-	K_VAR,
-	K_TYPE,         /* name of an OCaml type */
+typedef enum
+{
+	K_CLASS,  /* Ocaml class, relatively rare */
+	K_METHOD, /* class method */
+	K_MODULE, /* Ocaml module OR functor */
+	K_VARIABLE,
+	K_VAL,
+	K_TYPE, /* name of an OCaml type */
 	K_FUNCTION,
-	K_CONSTRUCTOR,  /* Constructor of a sum type */
+	K_CONSTRUCTOR, /* Constructor of a sum type */
 	K_RECORDFIELD,
 	K_EXCEPTION
 } ocamlKind;
 
 static kindOption OcamlKinds[] = {
-	{TRUE, 'c', "class", "classes"},
-	{TRUE, 'm', "method", "Object's method"},
-	{TRUE, 'M', "module", "Module or functor"},
-	{TRUE, 'v', "var", "Global variable"},
-	{TRUE, 't', "type", "Type name"},
-	{TRUE, 'f', "function", "A function"},
-	{TRUE, 'C', "Constructor", "A constructor"},
-	{TRUE, 'r', "Record field", "A 'structure' field"},
-	{TRUE, 'e', "Exception", "An exception"}
+	{ TRUE, 'c', "class", "classes" },
+	{ TRUE, 'm', "method", "Object's method" },
+	{ TRUE, 'M', "module", "Module or functor" },
+	{ TRUE, 'v', "variable", "Global variable" },
+	{ TRUE, 'p', "val", "Signature item" },
+	{ TRUE, 't', "type", "Type name" },
+	{ TRUE, 'f', "function", "A function" },
+	{ TRUE, 'C', "constructor", "A constructor" },
+	{ TRUE, 'r', "record_field", "A 'structure' field" },
```

Comment:
Ecb has a number of issues dealing with uppercase names and especially with tags that have spaces in them, that was the main reason for altering those tags. I also wanted to match closer to the c/c++/java tags, as it reduced the need to implement custom tag interpretation in ecb (which is significantly more messy to deal with).